test/events.js

```diff
@@ -22,6 +22,44 @@ describe('events', function () {
     })
   })
 
+  it('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return client.query('SELECT now()')
+      .then(() => {
+        emittedClient = client
+      })
+    })
+
+    pool.connect(function (err, client, release) {
+      if (err) return done(err)
+      release()
+      pool.end()
+      expect(client).to.be(emittedClient)
+      done()
+    })
+  })
+
+  it('catch error on connect', function (done) {
```

Comment:
please rename to catch error on connect listener.
the connection itself was done successfully in this test, the error was raised from the event listener

Reply:
renamed